returnn/torch/engine.py

```diff
@@ -51,7 +51,10 @@ def __init__(self, config: Config):
         self._save_model_epoch_interval = 1
         self._updater = None  # type: Optional[Updater]
 
-        self._device = "cuda" if torch.cuda.is_available() and torch.cuda.device_count() > 0 else "cpu"
+        self._device = "cuda" if config.typed_value("device") == "gpu" and (torch.cuda.is_available() and torch.cuda.device_count() > 0) else "cpu"
```

Comment:
```suggestion
        self._device = "cuda" if config.typed_value("device") == "gpu" else "cpu"
```

You can skip the check here for cuda availability and devices, as if this is not given anyway the assert will fail.